tests/unit/base/trained_model_serialization_unittest.cc.jinja2

```diff
@@ -0,0 +1,210 @@
+			}
+
+			for (index_t i=0; i<N_TEST; ++i)
+			{
+				X_test_reg(0,i)=CMath::random(-1.0, 1.0);
+				X_test_reg(1,i)=CMath::random(-1.0, 1.0);
+			}
+
+			feat_train_reg=new CDenseFeatures<float64_t>(X_reg);
+			feat_test_reg=new CDenseFeatures<float64_t>(X_test_reg);
+			label_train_reg=new CRegressionLabels(Y_reg);
+			SG_REF(feat_train_reg);
+			SG_REF(feat_test_reg);
+			SG_REF(label_train_reg);
+		}
+
+		static void TearDownTestCase() {
+			SG_UNREF(feat_train_bin);
+			SG_UNREF(feat_test_bin);
+			SG_UNREF(label_train_bin);
+
+			SG_UNREF(feat_train_reg);
+			SG_UNREF(feat_test_reg);
+			SG_UNREF(label_train_reg);
+		}
+
+		static const index_t N_TRAIN=30;
+		static const index_t N_TEST=10;
+
+		static CDenseFeatures<float64_t>* feat_train_bin;
+		static CDenseFeatures<float64_t>* feat_test_bin;
+		static CBinaryLabels* label_train_bin;
+
+		static CDenseFeatures<float64_t>* feat_train_reg;
+		static CDenseFeatures<float64_t>* feat_test_reg;
+		static CRegressionLabels* label_train_reg;
+};
+
+CDenseFeatures<float64_t>* TrainedModelSerializationTest::feat_train_bin;
+CDenseFeatures<float64_t>* TrainedModelSerializationTest::feat_test_bin=NULL;
+CBinaryLabels* TrainedModelSerializationTest::label_train_bin=NULL;
+
+CDenseFeatures<float64_t>* TrainedModelSerializationTest::feat_train_reg=NULL;
+CDenseFeatures<float64_t>* TrainedModelSerializationTest::feat_test_reg=NULL;
+CRegressionLabels* TrainedModelSerializationTest::label_train_reg=NULL;
+
+{% macro test(class, problem_type) -%}
+TEST_F(TrainedModelSerializationTest, {{class}})
+{
+{% if problem_type == 'binary' %}
+	CDenseFeatures<float64_t>* feat_train=feat_train_bin;
+	CDenseFeatures<float64_t>* feat_test=feat_test_bin;
+	CBinaryLabels* label_train=label_train_bin;
+{% elif problem_type == 'regression' %}
+	CDenseFeatures<float64_t>* feat_train=feat_train_reg;
+	CDenseFeatures<float64_t>* feat_test=feat_test_reg;
+	CRegressionLabels* label_train=label_train_reg;
+{% endif %}
+
+	/* instantiate and train the machine */
+	auto machine=new C{{class}}();
+	machine->set_features(feat_train);
+	machine->set_labels(label_train);
+	bool train_success=machine->train();
+	ASSERT_TRUE(train_success);
+
+	/* predict with trained machine */
+	CLabels* predictions=machine->apply(feat_test);
+//	SGVector<float64_t> prediction_vector=predictions->get_labels();
+
+	/* serialize the machine */
+	std::string class_name("{{class}}");
+	std::string file_template="/tmp/shogun-unittest-linear-machine-serialization-" + class_name + ".XXXXXX";
+	char* filename=mktemp_cst(const_cast<char*>(file_template.c_str()));
+
+	CSerializableAsciiFile *file=new CSerializableAsciiFile(filename, 'w');
+	bool save_success=machine->save_serializable(file);
+	file->close();
+	SG_UNREF(file);
+	ASSERT_TRUE(save_success);
+
+	/* deserialize the machine */
+	file=new CSerializableAsciiFile(filename, 'r');
+	auto deserialized_model=(C{{class}}*)new_sgserializable(class_name.c_str(), PT_NOT_GENERIC);
```

Comment:
I think you can use jinja to instantiate using the default constructor here
The `new_sgserializable` is quite ugly inside, so I would like to avoid using it. And you know the classname anyways here